xbmc/video/VideoDatabase.cpp

```diff
@@ -2097,15 +2097,19 @@ int CVideoDatabase::SetDetailsForMovie(const std::string& strFilenameAndPath, co
     }
     // update our movie table (we know it was added already above)
     // and insert the new row
-    std::string sql = "update movie set " + GetValueString(details, VIDEODB_ID_MIN, VIDEODB_ID_MAX, DbMovieOffsets);
+    std::string sql = "UPDATE movie SET " + GetValueString(details, VIDEODB_ID_MIN, VIDEODB_ID_MAX, DbMovieOffsets);
     if (idSet > 0)
```

Comment:
`PrepareSQL`, `StringUtils::Format` and `CLog::Log` can't handle `std::string`s but only `(const) char*`s so you would have to use
```
std::string CVideoDatabase::AddUpdateParameter(std::string name, int value)
{
  if (value > 0)
        return PrepareSQL(", %s = %i", name.c_str(), value);
      else
        return StringUtils::Format(", %s = NULL", name.c_str());
}
```